OracleUnifiedDirectory/imagetool/12.2.1.4.0/README.md

```diff
@@ -0,0 +1,433 @@
+[INFO   ] Added entry 28186730_13.9.4.2.2=/scratch/OUDDockerK8S/stage/p28186730_139422_Generic.zip
+$ ./imagetool.sh cache addEntry --key 31400392_12.2.1.4.0 --value /scratch/OUDDockerK8S/stage/p31400392_122140_Generic.zip
+[INFO   ] Added entry 31400392_12.2.1.4.0=/scratch/OUDDockerK8S/stage/p31400392_122140_Generic.zip
+$
+```
+
+### iv) Create the OUD image
+
+Execute the `imagetool create` command to create the OUD image.
+
+The following parameters are provided as input to the create command,
+
+* jdkVersion - JDK version to be used in the image.
+* type - type of image to be built.
+* version - version of the image.
+* tag - tag name for the image.
+* additionalBuildCommands - additional build commands provided as a text file.
+* addtionalBuildFiles - path of additional build files as comma separated list.
+
+
+Below a sample command used to build OUD image,
+
+```
+$ ./imagetool.sh create --jdkVersion=8u241 --type oud --version=12.2.1.4.0 \
+    --tag=oud-with-patch:12.2.1.4.0 \
+    --additionalBuildCommands <work directory>/docker-images/OracleUnifiedDirectory/dockerfiles/12.2.1.4.0/imagetool/12.2.1.4.0/buildCmds.txt \
+    --additionalBuildFiles <work directory>/docker-images/OracleUnifiedDirectory/dockerfiles/12.2.1.4.0/container-scripts \
+    --patches <patch_a>,<patch_b>,...
+```
+
+For example:
+
+```
+$ ./imagetool.sh create --jdkVersion=8u261 --type oud --version=12.2.1.4.0 \
+--tag=oud-with-patch:12.2.1.4.0 \
+--additionalBuildCommands /scratch/OUDDockerK8S/docker-images/OracleUnifiedDirectory/imagetool/12.2.1.4.0/additionalBuildCmds.txt \
+--additionalBuildFiles /scratch/OUDDockerK8S/docker-images/OracleUnifiedDirectory/dockerfiles/12.2.1.4.0/container-scripts \
+--patches 28186730,31400392
+```
+
+### v) View the docker image
+
+Run the `docker images` command to ensure the new OAM image is loaded into the repository:
+
+```
+$ docker images
+REPOSITORY                                                                   TAG                       IMAGE ID            CREATED             SIZE
+oud-with-patch                                                               12.2.1.4.0                8a937042bef3        9 minutes ago       992MB
+...
+```
+
+# 8. Sample Dockerfile generated with imagetool
+
+Below is a sample dockerfile created with the imagetool. This can be viewed by issuing the `imagetool` command with the `--dryRun` option:
+
+```
+./imagetool.sh @<work directory/build/buildArgs --dryRun
+```
+
+```
+	########## BEGIN DOCKERFILE ##########
+	#
+	# Copyright (c) 2019, 2020, Oracle and/or its affiliates.  All rights reserved.
+	#
+	# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+	#
+	#
+	FROM oraclelinux:7-slim as OS_UPDATE
+	LABEL com.oracle.weblogic.imagetool.buildid="f3557b16-8e34-4bf0-be6a-baa06fd8c137"
+	USER root
+	
+	RUN yum -y --downloaddir= install gzip tar unzip \
+	 && yum -y --downloaddir= clean all \
```

Comment:
You don't need both `yum clean all` and `rm -rf /var/cache/yum`. The latter is more efficient.

Reply:
Dockerfile content mentioned in this README is generated through imagetool. Imagetool seems to be including both. We will update this dryrun output in future with usage of updated imagetool.

Reply:
This is why I dislike using/relying on Image Tool to build your Dockerfile. You should know what the most efficient Dockerfile for your product looks like so that you can adjust Image Tool accordingly (or provide feedback to the Image Tool developers).

Reply:
For now, removed the output generated through imagetool.